integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/End2EndTestITStep.java

```diff
@@ -0,0 +1,257 @@
+                .then()
+                .statusCode(401);
+    }
+
+    @Given("^get list of Bridge instances with access token doesn't contain Bridge \"([^\"]*)\"$")
+    public void getEmptyBridges(String bridgeName) {
+        BridgeListResponse response = getBridgeList();
+        assertThat(response.getKind()).isEqualTo("BridgeList");
+        if (response.getItems().size() > 0) {
+            for (int i = 0; i < response.getItems().size(); i++) {
+                assertThat(response.getItems().get(i).getName()).isNotEqualTo(bridgeName);
+            }
+        }
+    }
+
+    @When("^create a Bridge with name \"([^\"]*)\" with access token$")
+    public void createBridge(String bridgeName) {
+        BridgeResponse response = addBridge(bridgeName);
+        assertThat(response.getName()).isEqualTo(bridgeName);
+        assertThat(response.getStatus()).isEqualTo(BridgeStatus.REQUESTED);
+        assertThat(response.getEndpoint()).isNull();
+        assertThat(response.getPublishedAt()).isNull();
+        assertThat(response.getHref()).isNotNull();
+        assertThat(response.getSubmittedAt()).isNotNull();
+
+        bridgeId = response.getId();
+    }
+
+    @Given("^get list of Bridge instances with access token contains Bridge \"([^\"]*)\"")
+    public void testBridgeExists(String bridgeName) {
+        BridgeListResponse response = getBridgeList();
+        List<String> bridgeNames = new ArrayList<>();
+        if (response.getItems().size() > 0) {
+            for (int i = 0; i < response.getItems().size(); i++) {
+                bridgeNames.add(response.getItems().get(i).getName());
+            }
+        }
+        assertThat(bridgeNames.contains(bridgeName)).isTrue();
+    }
+
+    @Then("^get Bridge with access token exists in status \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
+    public void bridgeDeployedWithinMinutes(String status, int timeoutMinutes) {
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutMinutes))
+                .pollInterval(Duration.ofSeconds(5))
+                .untilAsserted(
+                        () -> getBridgeDetails(bridgeId)
+                                .then()
+                                .body("status", Matchers.equalTo(status))
+                                .body("endpoint", Matchers.containsString(bridgeId)));
+
+        // store bridge endpoint details
+        endPoint = getBridgeDetails(bridgeId).then().extract().response().as(BridgeResponse.class).getEndpoint();
+    }
+
+    @When("^add Processor to the Bridge with access token:$")
+    public void addProcessor(String processorRequestJson) {
+        InputStream resourceStream = new ByteArrayInputStream(processorRequestJson.getBytes(StandardCharsets.UTF_8));
+        ProcessorResponse response = createProcessor(bridgeId, resourceStream);
+
+        processorId = response.getId();
+
+        assertThat(response.getName()).isEqualTo(PROCESSOR_NAME);
```

Comment:
This may lead into unexpected failures in future tests, we should either specify processor name as a parameter (and part of step) or parse it from request json.

Reply:
Yes agree..I will add this processor name and topic  as part of step parameter